extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php

```diff
@@ -44,9 +45,58 @@ static public function onParserTagHooksBeforeInvoke( $name, $marker, $content, $
 		return true;
 	}
 
-	static public function onWgQueryPages( &$queryPages = [ ] ) {
+	public static function onWgQueryPages( &$queryPages = [ ] ) {
 		$queryPages[] = [ 'AllinfoboxesQueryPage', 'AllInfoboxes' ];
 
 		return true;
 	}
+
+	/**
+	 * @param $article
+	 * @param $text
+	 * @param $wgOut
+	 *
+	 * @return bool
+	 */
+	public static function onShowMissingArticleBeforeTextAppend( &$article, &$text, &$wgOut, &$errors ) {
+		$title = $article->getTitle();
+
+		if (
+			$title &&
+			!$title->exists() &&
+			$title->getNamespace() === NS_TEMPLATE &&
+			!count( $errors )
+		) {
+			$text = '';
+
+			$HTML = F::app()->renderView(
+				'PortableInfoboxBuilderSpecialController',
+				'renderCreateTemplateEntryPoint',
+				[ 'title' => $title->getText() ]
+			);
+
+			$wgOut->addHTML( $HTML );
```

Comment:
to `$wgOut` we want to add HTML and if we change the `$text` variable then the we will get the `$wgOut->addWikiText( $text );` in `Article.php` so it will try to render the wikitext